modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.jsx

```diff
@@ -170,54 +187,69 @@ const AppContextProvider = ({children}) => {
 		};
 
 		const fetchData = async () => {
-			const projectExternalReferenceCode = getAccountKey();
-
-			if (!projectExternalReferenceCode) {
-				Liferay.Util.navigate(pageRoutes.home());
-			}
+			try {
```

Comment:
what is this new try/catch catching specifically?

Reply:
what happens when this try/catch catches an error that is not the 404, if another type of random javascript error happens and they should have access to the project?

Reply:
@samuelleolopes for the most part, try/catch should not be used for valid logic workflow, it should be only used for actual exception handling (actual errors, not valid permission checks). so i dont believe the try/catch is the best way to handle this. since jennny is taking over this ticket now you don't have to worry about fixing it, but something to remember for the future

Reply:
I understand your question, the try/catch was set specifically for when the user doesn't have access, Eduardo and I did several tests for this.

Reply:
If necessary, we can look for a more specific way of handling the error.